projects/packages/videopress/src/client/block-editor/hooks/use-video-player/index.ts

```diff
@@ -0,0 +1,150 @@
+	 * - - Detect the first time the player plays.
+	 * - - Stop right after it plays.
+	 * - - Set the player position at the desired time.
+	 *
+	 * @param {MessageEvent} event - Message event
+	 */
+	function listenEventsHandler( event: MessageEvent ) {
+		const { data: eventData = {}, source } = event;
+		const { event: eventName } = event?.data || {};
+
+		// Detect when the video has been loaded.
+		if ( eventName === 'videopress_loading_state' && eventData.state === 'loaded' ) {
+			debug( 'state: loaded' );
+			playerState.current = 'loaded';
+		}
+
+		// Detect when the video has been played for the first time.
+		if ( eventName === 'videopress_playing' && playerState.current === 'loaded' ) {
+			playerState.current = 'first-play';
+			debug( 'state: first-play detected' );
+
+			// Pause and move the video at the desired time.
+			source.postMessage( { event: 'videopress_action_pause' }, { targetOrigin: '*' } );
+
+			// Set position at time if it was provided.
+			if ( typeof atTime !== 'undefined' ) {
+				debug( 'set position at time %o ', atTime );
+				source.postMessage(
+					{ event: 'videopress_action_set_currenttime', currentTime: atTime / 1000 },
+					{ targetOrigin: '*' }
+				);
+			}
+
+			// Here we consider the video as ready to be controlled.
+			setPlayerIsReady( true );
+			playerState.current = 'ready';
+		}
+	}
+
+	// Listen player events.
+	useEffect( () => {
+		if ( isRequestingPreview ) {
+			return;
+		}
+
+		const sandboxIFrameWindow = getIframeWindowFromRef( iFrameRef );
+		if ( ! sandboxIFrameWindow ) {
+			return;
+		}
+
+		sandboxIFrameWindow.addEventListener( 'message', listenEventsHandler );
+
+		return () => {
+			// Remove the listener when the component is unmounted.
+			sandboxIFrameWindow.removeEventListener( 'message', listenEventsHandler );
+		};
+	}, [ iFrameRef, isRequestingPreview ] );
+
+	const playVideo = useCallback( () => {
+		const sandboxIFrameWindow = getIframeWindowFromRef( iFrameRef );
+		if ( ! sandboxIFrameWindow || ! playerIsReady ) {
+			return;
+		}
+
+		sandboxIFrameWindow.postMessage( { event: 'videopress_action_play' }, '*' );
+	}, [ iFrameRef, playerIsReady ] );
+
+	const stopVideo = useCallback( () => {
+		const sandboxIFrameWindow = getIframeWindowFromRef( iFrameRef );
+		if ( ! sandboxIFrameWindow || ! playerIsReady ) {
+			return;
+		}
+
+		sandboxIFrameWindow.postMessage( { event: 'videopress_action_pause' }, '*' );
+	}, [ iFrameRef, playerIsReady ] );
+
+	// PreviewOnHover feature.
+	useEffect( () => {
+		if ( ! wrapperElement || ! previewOnHover ) {
+			return;
+		}
+
+		wrapperElement.addEventListener( 'mouseenter', playVideo );
+		wrapperElement.addEventListener( 'mouseleave', stopVideo );
+
+		return () => {
+			// Remove the listener when the component is unmounted.
+			wrapperElement.removeEventListener( 'mouseenter', playVideo );
+			wrapperElement.removeEventListener( 'mouseleave', stopVideo );
+		};
+	}, [ previewOnHover, wrapperElement, playerIsReady ] );
```

Comment:
For me looks strange the hook is called `useVideoPlayer` and it implements the `onHover` feature. Looks it should return all instances and the poster panel applies the action.

Reply:
> The idea is to centralize the code that implements these features, but it's something that we can change in the short term

Gotcha. But maybe we should change the hook name? Or make the feature `opt-in`? Because right now it applied by default, right?

Reply:
Actually, it doesn't control the player by default. It requires some options to be defined for it.

> Or make the feature opt-in

Could you develop the idea a little bit more?

Reply:
> Could you develop the idea a little bit more?

I just checked again and missed that it is already `opt-in`. We have the `previewOnHover` flag. Missed that, sorry